src/seedu/addressbook/data/person/addressComponents/Block.java

```diff
@@ -0,0 +1,19 @@
+package seedu.addressbook.data.person.addressComponents;
+
+/**
+ * Represents a person's block number in their address
+ * Guarantees: blockNumber is immutable
+ *
+ */
+public class Block {
+
+    private final String blockNumber;
+
+    public Block(String blockNum) {
```

Comment:
Would adding another layer of validation be improve the implementation of this class?